tests/cache/tests.py

```diff
@@ -1221,6 +1221,12 @@ def test_creates_cache_dir_if_nonexistent(self):
         cache.set('foo', 'bar')
         os.path.exists(self.dirname)
 
+    def test_cache_write_unpickable_type(self):
+        try:
```

Comment:
Hi Tim,

I haven't thought of any upgrade considerations. What kind of tests should I write to ensure that there will be no problems when the user upgrades?

I revised the patch with all your comments. Thanks. I will submit the patch again after this one is sorted out.